tests/component-tests/textToSpeech/js/TextToSpeechTests.js

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+*/
+
+// Declare dependencies
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.tests");
+
+    fluid.defaults("fluid.tests.textToSpeech", {
+        gradeNames: ["fluid.textToSpeech", "autoInit"],
+        utteranceOpts: {
+            // not all speech synthesizers will respect this setting
+            volume: 0
+        }
+    });
+
+    // only run the tests in browsers that support the Web Speech API for speech synthesis
+    if (!fluid.textToSpeech.isSupported()) {
+        jqUnit.test("No Tests Run", function () {
+            jqUnit.assert("Does not support the SpeechSynthesis");
+        });
+
+    } else {
+        jqUnit.test("Initialization", function () {
+            var that = fluid.tests.textToSpeech();
+
+            jqUnit.assertTrue("The Text to Speech component should have initialized", that);
+            jqUnit.assertFalse("Nothing should be speaking", that.isSpeaking());
+            jqUnit.assertFalse("Nothing should be pending", that.isPending());
+            jqUnit.assertFalse("Shouldn't be paused", that.isPaused());
+        });
+
+        jqUnit.asyncTest("Start and Stop Events", function () {
+            jqUnit.expect(8);
+            var that = fluid.tests.textToSpeech({
```

Comment:
Make this into a proper grade as usual - in general avoid large amounts of in-line configuration in functions